[Partillay] iP

Partillay opened the pull request:
# Partillay

> Slaying in life is the best policy there is.

Partillay frees your mind of having to remember things you need to do. It's,
- Easy to use
- Text-based
- YOUR ABSOLUTELY ~~BESTIE~~ BEST CHATBOT

And of course it's **FREE**!

Features:
1. Managing tasks
2. View tasks
3. Mark and unmark tasks

## AND OF COURSE YOU CAN ENJOY:
- [ ] SEEING KIM SUNOO 😍 

### Wait, but what is a [Kim Sunoo](https://kprofiles.com/sunoo-enhypen-profile/)?
P.s. He is a SUNSHINE

## Heads up, you can add date formats that you like in the code!
You can easily do this in the enum `DateFormat`!

```java
public enum DateFormat {
    YYYY_MM_DD_HH_MM_1("yyyy-MM-dd HH:mm"),
    YYYY_MM_DD_HH_MM_2("yyyy/MM/dd HH:mm"),
    YYYY_MM_DD_HH_MM_3("yyyy-MM-dd HHmm"),
    YYYY_MM_DD_HH_MM_4("yyyy/MM/dd HHmm"),

    YYYY_MM_DD_1("yyyy-MM-dd"),
    YYYY_MM_DD_2("yyyy/MM/dd"),

    D_MM_YYYY("d/MM/yyyy"),
    DD_MM_YYYY("dd/MM/yyyy"),

    DD_MM_YYYY_HH_MM_1("dd-MM-yyyy HH:mm"),
    DD_MM_YYYY_HH_MM_2("dd/MM/yyyy HH:mm"),
    DD_MM_YYYY_HH_MM_3("dd-MM-yyyy HHmm"),
    DD_MM_YYYY_HH_MM_4("dd/MM/yyyy HHmm"),

    YYYY_MM_DD_T_HH_MM("yyyy-MM-dd'T'HH:mm");

    // Add more formats for ease of input if you like! //
    // ...
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Partillay791 committed:
Deleted/Rename original file and Added a greeting

Partillay791 committed:
Added echoes

Partillay791 committed:
added support for add,list

Partillay791 committed:
Revert "added support for add,list"

This reverts commit 8619905df43a2d9dd5678d5c72e6c44e0c7135d2.

Partillay791 committed:
Reapply "added support for add,list"

This reverts commit 204f0479aec2d7f0f2d9cd430a45c1b1cc627ea6.

Partillay791 committed:
Added Mark as Done

Partillay791 committed:
Fixed Level-3

Partillay791 committed:
Added ToDo, Event, Deadline

Partillay791 committed:
Added A-TextUiTesting

Partillay791 committed:
Handled some Errors

Partillay791 committed:
Added Delete

jordon0802 commented on src/main/java/Partillay.java:
```diff
@@ -0,0 +1,246 @@
+                    }
+                    description = description.trim();
+                    if (description.isEmpty()) {
+                        throw new PartillayException("Bestie, your task cannot be empty!");
```
Name of the exception could perhaps be more descriptive?

jordon0802 commented on src/main/java/Partillay.java:
```diff
@@ -0,0 +1,246 @@
+import java.util.Scanner;
+
+public class Partillay {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
```
Like that you have a constant for a horizontal line following the naming convention for constants!

jordon0802 commented on src/main/java/Partillay.java:
```diff
@@ -0,0 +1,246 @@
+
+                // Handle an error here (index out of bounds)
+                if (userInput.startsWith("mark")) {
+                    int taskIndexToMark = Integer.parseInt(userInput.substring(5));
```
Very clear and descriptive variable name!!

jordon0802 commented on src/main/java/Partillay.java:
```diff
@@ -0,0 +1,246 @@
+        System.out.println("Anything I CAN HELP?");
+        System.out.println(HORIZONTAL_LINE);
+
+        Scanner sc = new Scanner(System.in);
```
Though its common to name the scanner sc, but perhaps this could be named more semantically? (e.g. scanner)

jordon0802 reviewed (commented):
Overall, I find your code easy to read, considering that most of your variable and class names are semantical!! Just a few nits😬

johngao122 commented on src/main/java/Partillay.java:
```diff
@@ -16,79 +16,141 @@ public static void main(String[] args) {
+                // Handle an error here (index out of bounds)
+                if (userInput.startsWith("unmark")) {
+                    int taskIndexToUnmark = Integer.parseInt(userInput.substring(7));
+                    if (taskIndexToUnmark > taskIndex) {
```
Nice error handling!

johngao122 commented on src/main/java/Partillay.java:
```diff
@@ -16,79 +16,141 @@ public static void main(String[] args) {
+                    taskIndex++;
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(tasks[taskIndex - 1]);
+                    System.out.println("Now you have " + taskIndex + " tasks in the list.");
```
Perhaps extract these into UI helper methods?

johngao122 commented on src/main/java/Partillay.java:
```diff
@@ -16,79 +16,141 @@ public static void main(String[] args) {
-                System.out.println("Here are the tasks in your list:");
-                for (int i = 0; i < taskIndex; i++) {
-                    System.out.println(i + 1 + "." + tasks[i]);
+                if (userInput.equals("bye")) {
```
Have you considered using an enum for command types instead of string comparisons? It could make the code more maintainable.

johngao122 reviewed (commented):
I like the overall implementation in this PR, particularly the robust input validation and descriptive error messages that enhance the user experience. The code handles edge cases well and maintains a consistent, friendly tone in the UI.

Partillay791 committed:
Separate and create new classes

Partillay791 committed:
Add new constructors to read from txt file

Partillay791 committed:
Add saving function

Partillay791 committed:
Merge branch 'branch-Level-7'

Partillay791 committed:
Add DateTimeFormatParser class

Partillay791 committed:
Merge branch 'branch-Level-8'

Partillay791 committed:
Add a TaskList class

Partillay791 committed:
Merge branch 'branch-A-MoreOOP'

Partillay791 committed:
Modify directory and packages

Partillay791 committed:
Merge branch 'branch-A-Packages'

Partillay791 committed:
Merge branch 'add-gradle-support'

Partillay791 committed:
Update build.gradle

Partillay791 committed:
Add equals method to facilitate test

Partillay791 committed:
Add JUnit Tests

Partillay791 committed:
Merge branch 'branch-A-JUnit'

Partillay791 committed:
Create a JAR file

Partillay791 committed:
Merge branch 'branch-A-Jar'

Partillay791 committed:
Add JavaDoc comments to half the classes/methods

Partillay791 committed:
Modify code to follow coding standard

Partillay791 committed:
Add Find method and modify Task class

Partillay791 committed:
Merge branch 'branch-A-JavaDoc'

Partillay791 committed:
Merge branch 'branch-A-CodingStandard'

Partillay791 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/partillay/task/Task.java

Partillay791 committed:
Add checkStyle for coding style violation detection

Partillay791 committed:
Merge branch 'branch-A-Checkstyle'

Partillay791 committed:
Add Launcher and Main class

Partillay791 committed:
Apply basic echo function in gui

Partillay791 committed:
Update JavaDoc for exception classes

Partillay791 committed:
Update command classes for string output

Partillay791 committed:
Update gui for chatbot

Partillay791 committed:
Merge branch 'branch-Level-10'

Partillay791 committed:
Edit throwing exception condition

Partillay791 committed:
Add assertions to execute methods

Currently there are no assertions.

Assertions are added to execute methods of AddCommand and DeleteCommand
classes to check internal logic.

Precisely,
* AddCommand: task size should increment by 1 upon execution
* DeleteCommand: task size should decrement by 1 upon execution

Partillay791 committed:
Improve code quality of Parser class

The parse method is too long (i.e. 118 lines).

This causes the code to be potentially hard to read.

SLAP abstraction mechanism is applied.

This can effectively shorten the code.

Multiple helper methods (e.g. parseTaskIndex) are defined to help
separate the original code into simpler portions while maintaining its
functionality.

Partillay committed:
Merge pull request #2 from Partillay/branch-A-Assertions

Branch a assertions

Partillay791 committed:
Merge branch 'master' of https://github.com/Partillay/ip

Partillay791 committed:
Merge branch 'master' of https://github.com/Partillay/ip

Partillay791 committed:
Merge branch 'master' into branch-A-CodeQuality

Partillay committed:
Merge pull request #3 from Partillay/branch-A-CodeQuality

Improve code quality of Parser class

Partillay791 committed:
Merge branch 'master' of https://github.com/Partillay/ip

Partillay791 committed:
Add friendlier syntax extension

Partillay791 committed:
Update gui of chatbot

Partillay791 committed:
Fix writing files to txt file

Partillay791 committed:
Add title to Partillay gui

Partillay791 committed:
Modify input normalization rules

Partillay791 committed:
Update current tasklist in txt file

Partillay791 committed:
Add representative screenchot of product

Partillay791 committed:
Fix find command to search full words

Partillay791 committed:
Fix event class to handle proper start times

Partillay791 committed:
Update user guide for chatbot GUI application

Partillay791 committed:
Update user guide with date time formats that can be accepted

Partillay791 committed:
Add method to compare precedence of date time of strings

Partillay791 committed:
Increase error message specificity and prevent wrong datetime logic
